runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2184,7 +2184,7 @@ J9::Z::TreeEvaluator::asynccheckEvaluator(TR::Node * node, TR::CodeGenerator * c
          }
       if (value >= MIN_IMMEDIATE_VAL && value <= MAX_IMMEDIATE_VAL)
          {
-         TR::MemoryReference * tempMR = generateS390MemoryReference(firstChild, cg);
+         TR::MemoryReference * tempMR = TR::MemoryReference::create(cg, firstChild);
```

Comment:
@0xdaryl @fjeremic The rest of this PR looks fine to me but I don't have the background to tell if this is correct.  Can you of you +1 this?

Reply:
For the background, we got rid of calls to `generateS390MemoryReference(TR::Node *, TR::CodeGenerator*)` and replaced them with API `TR::MemoryReference::create` , not sure when and how this got escaped, may be because we do not have those changes in master branch of OpenJ9.

Reply:
This looks to be fine for the snapshot branch, but that API needs to find its way into the master branch still.